pyquil/paulis.py

```diff
@@ -261,11 +277,20 @@ def term_with_coeff(term, coeff):
     :returns: A new PauliTerm that duplicates term but sets coeff
     :rtype: PauliTerm
     """
-    new_pauli = copy.copy(term)
+    new_pauli = term.copy()
     new_pauli.coefficient = coeff
     return new_pauli
 
 
+def iszero(a, atol=1e-8):
```

Comment:
When I profiled the code w/ np.isclose, it slowed down the code immensely (over 2-4x slower) - actually proving a barrier for large fermionic operations (c.f. Nick). I could in-line the isclose functionality, but it seemed cleaner to just write a separate function.